articles/introduction-to-reinforcement-learning/index.md

```diff
@@ -0,0 +1,179 @@
+
+We will be going through these algorithms in a bit more depth.
+
+#### Q-learning
+Q-learning is the most popular among the RL algorithms.
+
+This algorithm finds an optimal action-selection policy by using a Q-function. By selecting the best of all possible actions, the algorithm is able to maximize the value function, Q.
+
+![q-learning](/engineering-education/introduction-to-reinforcement-learning/q-learning.PNG)
+
+*[Image Source: Springer Link](https://link.springer.com/chapter/10.1007/978-981-15-4095-0_2)*
+
+The Q-function takes in the state (s) and action (a) as its inputs.
+
+The Q-table stores the resulting Q values obtained from the output of the Q-function.
+
+The first action performed by the agent is completely randomized. This is because the Q-values are set to zero in the beginning.
+
+Values in the Q-table are then updated for each state-action pair. The process continues until the function Q converges to Qnew.
+
+**Discount factor**, usually denoted by the gamma parameter, this is a real value ranging from 0 to 1. It tells how important future rewards are to the current state.
+
+If the discount factor is set to 0.1, the agent will only learn about actions that produce immediate rewards. But, setting it to 0.9 will prompt the agent to check each of its actions based on the sum total of all its future rewards.
+
+**Learning rate (lr)** is also referred to as the step size. It determines how much new information it accepts over the old information.
+
+There is a need to balance the agent's exploration and exploitation efforts. This is because the agent must take random actions which ought to bring greater returns in the long run.
+
+Without this balance, the agent might opt for an immediate reward every time. The [epsilon-greedy algorithm](https://www.geeksforgeeks.org/epsilon-greedy-algorithm-in-reinforcement-learning/) achieves this balance. The agent will exploit the knowledge it has already gathered instead of exploring.
+
+That's Q-learning in a nutshell.
+
+#### Policy Gradients (PG)
+Alongside Q-learning, the Policy Gradient method is another type of reinforcement learning technique.
+
+It relies on optimizing policies in the long term cumulative reward.
+
+They estimate their values on an infinite number of states and actions. This makes them suitable for handling high dimensional continuous action spaces.
+
+They are of two types:
+
+**Stochastic Policy Gradients (SPG)**
+They integrates over both the state and action spaces.
+
+**Deterministic Policy Gradients (DPG)**
+They integrates over only the state spaces.
+
+Due to these differences, SPG may need more samples to compute in cases of high dimensions action spaces.
+
+#### Actor-Critics (AC)
```

Comment:
This whole AC section was a bit underwhelming - as a reader with little to no background on these topics - I found it quite difficult to follow along or understand what AC does ? or how it works. 
I'd recommend explaining a few of the concepts a bit more and in a manner that is simple enough for a beginner to follow.